File: OpenApi2.0/push-openapi-php-demo/QueryPushStatByMsg.php
<?php

include_once '../aliyun-php-sdk-core/Config.php';
include 'Push/Request/V20160801/QueryPushStatByMsgRequest.php';
use \Push\Request\V20160801 as Push;

$accessKeyId = "your accessKeyId";
$accessKeySecret = "your accessKeySecret";
$appKey = "your appKey";

$iClientProfile = DefaultProfile::getProfile("cn-hangzhou", $accessKeyId, $accessKeySecret);
$client = new DefaultAcsClient($iClientProfile);
$request = new Push\QueryPushStatByMsgRequest();

$request->setAppKey($appKey);
$request->setMessageId('500345');

$response = $client->getAcsResponse($request);
print_r("\r\n");
print_r($response);

?>

File: OpenApi2.0/push-openapi-net-demo/QueryPushStatByMsg.cs
using Aliyun.Acs.Core;
using Aliyun.Acs.Core.Exceptions;
using Aliyun.Acs.Core.Profile;
using Aliyun.Acs.Push.Model.V20160801;
using System;

namespace AlibabaCloud
{
    class QueryPushStat
    {
        static void Main()
        {
            IClientProfile clientProfile = DefaultProfile.GetProfile("cn-hangzhou", "<your access key id>", "<your access key secret>");
            DefaultAcsClient client = new DefaultAcsClient(clientProfile);
            QueryPushStatByMsgRequest request = new QueryPushStatByMsgRequest();
            request.AppKey = <your appKey>; 
            request.MessageId = "<MessageId>";
            
            try
            {
                QueryPushStatByMsgResponse response = client.GetAcsResponse(request);
                Console.WriteLine("RequestId:" + response.RequestId);
                foreach (QueryPushStatByMsgResponse.PushStat stat in response.PushStats)
                {
                    Console.WriteLine("MessageId:" + stat.MessageId);
                    Console.WriteLine("SentCount:" + stat.SentCount);
                    Console.WriteLine("ReceivedCount:" + stat.ReceivedCount);
                    Console.WriteLine("OpenedCount:" + stat.OpenedCount);
                    Console.WriteLine("DeletedCount:" + stat.DeletedCount);
                }
                Console.ReadLine();
            }
            catch (ServerException e)
            {
                Console.WriteLine(e.ErrorCode);
                Console.WriteLine(e.ErrorMessage);
                Console.ReadLine();
            }
            catch (ClientException e)
            {
                Console.WriteLine(e.ErrorCode);
                Console.WriteLine(e.ErrorMessage);
                Console.ReadLine();
            }
        }

    }
} 

